### ISSUE TITLE
netref dict does not expose .items() .keys() etc.

### ISSUE THREAD

**Issue by dirkroorda** [tomerfiliba/rpyc#272]
Thanks for this nice library.

The documentation on readthedocs documents 3.5, but the latest available release is 3.4.4.
That causes some avoidable friction in my head as a newcomer!

I got everything working, finally, but I stumbled on this: when my client gets a dict back from the server, as a netref, I can use the usual dict operations: `data[key]`, `for key in data:`, `data.__dict__`
But the methods with plain names do **not** work: `data.items()`. These attributes are not accessible.

I think it has to do with the access policy of attributes, but it is unclear from the docs how to configure the client (or the server? or the connection?) to let those attributes pass.

As a work-around I actually fetch such dictionaries to the client, by saying

`data = {key: data[key] for key in data}`

By the way, the dict in question is not an attribute of the .root object wrapped by the rpyc connection, but it is the result of a method that executes on the server.

Apart from this, everything works smoothly.

**Comment by coldfix** [comment]
> The documentation on readthedocs documents 3.5, but the latest available release is 3.4.4.
That causes some avoidable friction in my head as a newcomer!

Yes, very sorry, I was hoping to release after one or two weeks but then didn't have the time.

> But the methods with plain names do not work: data.items

I assume you're exposing a non-*classic* service. In this case, you'd have to pass something like `protocol_config={'allow_public_attrs': True}` when creating the server instance. For more config options, see http://rpyc.readthedocs.io/en/latest/api/core_protocol.html#rpyc.core.protocol.DEFAULT_CONFIG.

Unfortunately, it is currently not easy to adjust access policies for certain objects such as builtin dict/list instances. In fact, you could implement it yourself by subclassing or monkeypatching `Connection._access_attr`)

I agree that it can be confusing that you can't access public attributes from builtins such as list/dict once you get hold of them. The underlying problem is that it's not easy to decide which attributes/methods are harmless and which should be limited access. Therefore, the default is to allow only explicitly enabled (`exposed_XXX`) attributes. However, I'm open to integrate another way to shoot oneself in the foot by simplifying to add your own policy if anybody provides a clean patch.

Best, Thomas

**Comment by dirkroorda** [comment]
Thomas,
thanks for your explanation. I hope you find the time and gusto to make 3.5.
I saw the docs about passing `allow_public_attrs` to the Connection object, but the missing bit was exactly how to do that. You just filled that bit in:

```python
ThreadedServer(TfService, port=port, protocol_config={'allow_public_attrs': True})
```
it worked!

Great, thanks again,
Dirk
